### DISCUSSION TITLE
課題14についての質問

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#54] [Q&A]
①14-2について質問です。
要素数10、値域が1~50のコードを作成することが難しくどうやって作るのかわかりませんでした。
これまでは、課題13のようにCommon13というページの中から関数を呼びだしたりして、CommonとPracticeの2つのページを使用し、コンソールに出力していました。
しかし、今回はCommonはなくPractice14の中だけで作成するため、その時のようにはいかないなと思い、どう作成するのかわかりませんでした。
今のところ、課題13を参考に作成を進めております。

ちなみに、課題14の問題1は、ネットで素数の作成を検索（素数  プログラムと検索しました）し、
https://noumenon-th.net/programming/2018/04/30/prime01/
上記のリンク元の真似をしました。
関数を作成し、呼び出すことやfor文の式については理解しました。
しかし、自分で作成していないので理解が十分ではないと思います。
ただ、コードを見て理解していました。

**Comment by nsc-kiwamu** [comment]
①14-2
Common13.tsのmakeRandomListという関数はPractice13.tsからじゃないと呼べないということはありません。
文面を見る限り、Common14.tsにmakeRandomListがあれば、課題13と同じように呼べるが
Common14.tsが無いので、Practice14.tsでCommon13.tsのmakeRandomList相当の処理を作成していうように見えます。

それでも作成はできますが、Practice14.tsからCommon13.tsのmakeRandomListを呼べば、作成できます。

↓は報告内容が不十分なので「相手に伝わるか」もう少し見直しましょう。
ソースコードの内容を相手に説明できるかどうか？
説明できないのであれば、理解が足りないと考えて下さい。
```
関数を作成し、呼び出すことやfor文の式については理解しました。
→for文は二重ループになっていますが、それぞれ理解できているでしょうか？
しかし、自分で作成していないので理解が十分ではないと思います。
→具体的にどこが十分ではないのでしょうか？
ただ、コードを見て理解していました。
「何をどこまで」理解したのでしょうか？
```
